### ISSUE TITLE
"docker ps" doesn't work after machine restart in swarm mode

### ISSUE THREAD

**Issue by groyee** [docker/docker#25857]
**Output of `docker version`:**

Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:05:53 2016
 OS/Arch:      linux/amd64

Server:
 Version:      swarm/1.2.4
 API version:  1.22
 Go version:   go1.5.4
 Git commit:   5d5f7f0
 Built:        Thu Jul 28 19:52:54 UTC 2016
 OS/Arch:      linux/amd64

**Output of `docker info`:**

Containers: 30
 Running: 26
 Paused: 0
 Stopped: 4
Images: 96
Server Version: swarm/1.2.4
Role: replica
Primary: 192.168.0.8:3376
Strategy: spread
Filters: health, port, containerslots, dependency, affinity, constraint
Nodes: 9
 batchprocessing1: 192.168.0.17:2376
  └ ID: KZN7:CK5H:DSXN:U7IU:3BPY:WDQ5:EKOT:JE6M:ERXL:TSAW:UO6F:UIK2
  └ Status: Healthy
  └ Containers: 2 (2 Running, 0 Paused, 0 Stopped)
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 14.38 GiB
  └ Labels: kernelversion=4.4.0-34-generic, machinerole=batchprocessing, machinetype=C, operatingsystem=Ubuntu 16.04.1 LTS, provider=azure, storagedriver=aufs
  └ UpdatedAt: 2016-08-18T23:37:17Z
  └ ServerVersion: 1.12.1
 batchprocessing2: 192.168.0.13:2376
  └ ID: JN2D:WU2T:ODUB:FYFC:WVG7:STAC:CVJN:ANS6:7WHN:MHL4:YEMC:ULGH
  └ Status: Healthy
  └ Containers: 2 (2 Running, 0 Paused, 0 Stopped)
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 14.38 GiB
  └ Labels: kernelversion=4.4.0-34-generic, machinerole=batchprocessing, machinetype=C, operatingsystem=Ubuntu 16.04.1 LTS, provider=azure, storagedriver=aufs
  └ UpdatedAt: 2016-08-18T23:36:46Z
  └ ServerVersion: 1.12.1
 kafka1: 192.168.0.9:2376
  └ ID: NMOP:72DR:IWCA:LNNK:UMBO:QO2S:DQKC:NLVJ:RY3V:P6FN:ZLRI:MYGQ
  └ Status: Healthy
  └ Containers: 4 (4 Running, 0 Paused, 0 Stopped)
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 7.145 GiB
  └ Labels: containerslots=4, kernelversion=4.2.0-41-generic, machinetype=A, operatingsystem=Ubuntu 15.10, provider=azure, storagedriver=aufs
  └ UpdatedAt: 2016-08-18T23:37:26Z
  └ ServerVersion: 1.12.0
 kafka2: 192.168.0.10:2376
  └ ID: 2FL5:AY3V:3DTY:Z7KE:PPOO:LH3K:EDKV:FXZV:6BX7:LPNI:H3YU:EQCD
  └ Status: Healthy
  └ Containers: 4 (4 Running, 0 Paused, 0 Stopped)
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 7.145 GiB
  └ Labels: containerslots=4, kernelversion=4.2.0-41-generic, machinetype=A, operatingsystem=Ubuntu 15.10, provider=azure, storagedriver=aufs
  └ UpdatedAt: 2016-08-18T23:37:27Z
  └ ServerVersion: 1.12.0
 kafka3: 192.168.0.11:2376
  └ ID: YDWS:CSP6:3RQJ:O5EP:UTA6:LMUU:HFOY:VRCH:VC5C:NHYB:WQSW:JMLH
  └ Status: Healthy
  └ Containers: 4 (4 Running, 0 Paused, 0 Stopped)
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 7.145 GiB
  └ Labels: containerslots=4, kernelversion=4.2.0-41-generic, machinetype=A, operatingsystem=Ubuntu 15.10, provider=azure, storagedriver=aufs
  └ UpdatedAt: 2016-08-18T23:36:46Z
  └ ServerVersion: 1.12.0
 mysql1: 192.168.0.16:2376
  └ ID: SPM2:2WA7:4IPG:YWSF:YCVV:HOYZ:AMCO:MWNR:EHZD:EVNL:UFSL:XPZ6
  └ Status: Healthy
  └ Containers: 2 (2 Running, 0 Paused, 0 Stopped)
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 7.144 GiB
  └ Labels: containerslots=4, kernelversion=4.4.0-34-generic, machinerole=mysql, machinetype=A, operatingsystem=Ubuntu 16.04.1 LTS, provider=azure, storagedriver=aufs
  └ UpdatedAt: 2016-08-18T23:37:17Z
  └ ServerVersion: 1.12.1
 swarm-master2: 192.168.0.8:2376
  └ ID: 7BWV:3KOJ:6HLS:HW2S:L3SH:NNYN:Q4AB:TEIJ:TVRL:RFYS:TOJQ:T7XM
  └ Status: Healthy
  └ Containers: 4 (4 Running, 0 Paused, 0 Stopped)
  └ Reserved CPUs: 0 / 1
  └ Reserved Memory: 0 B / 1.718 GiB
  └ Labels: containerslots=3, kernelversion=4.2.0-41-generic, operatingsystem=Ubuntu 15.10, provider=azure, storagedriver=aufs
  └ UpdatedAt: 2016-08-18T23:37:35Z
  └ ServerVersion: 1.12.0
 ws1: 192.168.0.14:2376
  └ ID: MZX7:U5AU:LWVE:I45S:DXQO:NUJQ:O4KE:EIFZ:OBZQ:TJ7Z:PNLG:COAV
  └ Status: Healthy
  └ Containers: 4 (0 Running, 0 Paused, 4 Stopped)
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 7.145 GiB
  └ Labels: kernelversion=4.2.0-41-generic, machinerole=ws, machinetype=B, operatingsystem=Ubuntu 15.10, provider=azure, storagedriver=aufs
  └ UpdatedAt: 2016-08-18T23:37:29Z
  └ ServerVersion: 1.12.0
 ws2: 192.168.0.15:2376
  └ ID: QIE5:TBKZ:4OMM:TQX4:DUHC:23SG:W2K2:6SQN:W4OG:4YE2:X6AK:Q6W4
  └ Status: Healthy
  └ Containers: 4 (4 Running, 0 Paused, 0 Stopped)
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 7.145 GiB
  └ Labels: kernelversion=4.2.0-41-generic, machinerole=ws, machinetype=B, operatingsystem=Ubuntu 15.10, provider=azure, storagedriver=aufs
  └ UpdatedAt: 2016-08-18T23:37:21Z
  └ ServerVersion: 1.12.0
Plugins:
 Volume:
 Network:
Swarm:
 NodeID:
 Is Manager: false
 Node Address:
Security Options:
Kernel Version: 4.2.0-41-generic
Operating System: linux
Architecture: amd64
CPUs: 17
Total Memory: 73.34 GiB
Name: d80d396d7b83
Docker Root Dir:
Debug Mode (client): false
Debug Mode (server): false
WARNING: No kernel memory limit support

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. Configure swarm cluster
2. Point to the swarm master: **_eval $(docker-machine env --swarm [YOUR_SWARM_MASTER])_**
3. Put some docker container parametrs into docker-compose.yml. It doesn't really matter. You can just put a basic startup of Redis in the docker-compose.yml. Configure it with: **_restart: always_**
4. Run _**docker-compose up -d**_
5. Run **_docker ps_**. You will see the container running.
6. Run **_docker-compose ps_**. You will see the container running.
7. Restart the machine where this container is running. **docker-machine restart [MACHINE_NAME]**

Now comes the problem:

**Describe the results you received:**

While the machine is down. **_docker ps_** doesn't show the running container which is fine because it is indeed not running. However, when the machine is up and running after the restart, the container is started again automatically but **_docker ps_** will still not show it.

The problem is even more serious. While the machine is down **_docker-compose ps_** doesn't work because it fails to connect to the failed machine. Which is I guess another defect I should open for docker-compose. Instead of listing what it can show it just shows:

**_Get https://192.168.0.14:2376/containers/8a086e626e16d95ed010faa0e8440317c739fff1ae50572d758466884823c833/json: dial tcp 192.168.0.14:2376: getsockopt: connection refused_**

So because one machine is down, **_docker-compose ps_** doesn't work at all. docker ps is working fine. 

However, when the machine restarts then **_docker ps_** doesn't show the running container and **_docker-compose ps_** does.

Bottom line is that in a real production system you can't trust the ps command. **_docker ps_** will always work but can show you a wrong status, **_docker-compose ps_** can just not work when one of the machines in the cluster is down.

**Describe the results you expected:**

1. **_docker-compose ps_** should show the running containers even if it can't connect to one of the machines. It should show what it can. Just like _**docker-machine ls**_ shows all running machines even if it can't connect to some of them.

2. **_docker ps_** should show the running containers after machine reboot.

**Additional information you deem important (e.g. issue happens only occasionally):**

If one of them would work that would be fine. But when both of them are not working this is a problem.

**Comment by thaJeztah** [comment]
This looks like an issue with Swarm (https://github.com/docker/swarm) not Swarm _mode_ (https://docs.docker.com/engine/swarm/) which was introduced in 1.12 (yes, the naming is confusing). This issue may be better opened in the Docker Swarm issue tracker https://github.com/docker/swarm/issues

**Comment by groyee** [comment]
Thanks. Will open it there.
